src/megaclient.cpp

```diff
@@ -943,12 +955,15 @@ void MegaClient::exec()
                     if (Waiter::ds - fc->urltime > 600)
                     {
                         // fetches pending for this unconnected channel - dispatch fresh connection
+                        LOG_debug << "Getting fresh download URL";
+                        fc->timeout.reset();
                         reqs.add(new CommandGetFA(this, cit->first, fc->fahref, httpio->chunkedok));
```

Comment:
Ensure that the timeout is disabled while the download URL is being received from API servers.